hpx/runtime/serialization/detail/preprocess.hpp

```diff
@@ -72,11 +66,10 @@ namespace hpx { namespace serialization { namespace detail
                 std::lock_guard<mutex_type> l(mtx_);
                 ++num_futures_;
             }
-            std::shared_ptr<future_await_container> this_(this->shared_from_this());
             future_data.set_on_completed(
-                [this_]()
+                [this]()
```

Comment:
Am 02.09.2016 5:51 nachm. schrieb "Hartmut Kaiser" <notifications@github.com
>:
>
> In hpx/runtime/serialization/detail/preprocess.hpp:
>
> > future_data.set_on_completed( > - [this_]() > + [this]()
>
> How is this kept alive?

In the parcel_await itself, which holds the "container" as well as the
archive as member and is kept alive long enough.

>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.